toolchain/check/handle_array.cpp

```diff
@@ -30,26 +30,26 @@ auto HandleArrayExpression(Context& context, Parse::Node parse_node) -> bool {
     return context.TODO(parse_node, "HandleArrayExpressionWithoutBounds");
   }
 
-  auto bound_node_id = context.node_stack().PopExpression();
+  SemIR::InstId bound_inst_id = context.node_stack().PopExpression();
   context.node_stack()
       .PopAndDiscardSoloParseNode<Parse::NodeKind::ArrayExpressionSemi>();
-  auto element_type_node_id = context.node_stack().PopExpression();
-  auto bound_node = context.nodes().Get(bound_node_id);
-  if (auto literal = bound_node.TryAs<SemIR::IntegerLiteral>()) {
+  SemIR::InstId element_type_inst_id = context.node_stack().PopExpression();
```

Comment:
Preserve `auto` for now:
```suggestion
  auto element_type_inst_id = context.node_stack().PopExpression();
```